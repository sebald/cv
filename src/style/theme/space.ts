export default {
  baseline: 16,
};
