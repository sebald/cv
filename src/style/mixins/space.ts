import flow from 'lodash-es/flow';
import isNumber from 'lodash-es/isNumber';

import { StyledMixin, ThemedStyledProps } from '../theme';
import { pxToRem } from '../utils';

const getBaseline = (props: ThemedStyledProps) => props.theme.space.baseline;

export const baseline: StyledMixin = flow(getBaseline, pxToRem);

export const padding = (
  top: number,
  right?: number,
  bottom?: number,
  left?: number
): StyledMixin => props => ({
  padding: [top, right, bottom, left]
    .filter(isNumber)
    .map(v => v * getBaseline(props))
    .map(v => pxToRem(v))
    .join(' '),
});
