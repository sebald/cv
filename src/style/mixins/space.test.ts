import { theme } from '..';
import { baseline, padding, space } from './space';

test('basline', () => {
  expect(baseline({ theme })).toMatchSnapshot();
});

test('padding', () => {
  expect(padding(1)({ theme })).toMatchSnapshot();
  expect(padding(1, 1, 1, 1)({ theme })).toMatchSnapshot();
  expect(padding(1, 2)({ theme })).toMatchSnapshot();
  expect(padding(1, 1, 2)({ theme })).toMatchSnapshot();
  expect(padding(1, 1, 2, 3)({ theme })).toMatchSnapshot();
});

test('spacing', () => {
  expect(space({ theme })).toEqual({});

  expect(space({ theme, mx: 1 })).toMatchSnapshot();
  expect(space({ theme, my: 1 })).toMatchSnapshot();
  expect(space({ theme, mx: 2, my: 1 })).toMatchSnapshot();
  expect(space({ theme, m: 2 })).toMatchSnapshot();
  expect(space({ theme, mx: 1, m: 2 })).toMatchSnapshot();
  expect(space({ theme, mx: 1, m: 3, my: 2 })).toMatchSnapshot();
});
