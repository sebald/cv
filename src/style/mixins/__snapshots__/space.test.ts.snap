// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basline 1`] = `"1rem"`;

exports[`padding 1`] = `
Object {
  "padding": "1rem",
}
`;

exports[`padding 2`] = `
Object {
  "padding": "1rem 1rem 1rem 1rem",
}
`;

exports[`padding 3`] = `
Object {
  "padding": "1rem 2rem",
}
`;

exports[`padding 4`] = `
Object {
  "padding": "1rem 1rem 2rem",
}
`;

exports[`padding 5`] = `
Object {
  "padding": "1rem 1rem 2rem 3rem",
}
`;

exports[`spacing 1`] = `
Object {
  "marginLeft": "1rem",
  "marginRight": "1rem",
}
`;

exports[`spacing 2`] = `
Object {
  "marginBottom": "1rem",
  "marginTop": "1rem",
}
`;

exports[`spacing 3`] = `
Object {
  "marginBottom": "1rem",
  "marginLeft": "2rem",
  "marginRight": "2rem",
  "marginTop": "1rem",
}
`;

exports[`spacing 4`] = `
Object {
  "margin": "2rem",
}
`;

exports[`spacing 5`] = `
Object {
  "margin": "2rem",
  "marginLeft": "1rem",
  "marginRight": "1rem",
}
`;

exports[`spacing 6`] = `
Object {
  "margin": "3rem",
  "marginBottom": "2rem",
  "marginLeft": "1rem",
  "marginRight": "1rem",
  "marginTop": "2rem",
}
`;
